src/tss2-tcti/tcti-device.c

```diff
@@ -422,7 +422,7 @@ Tss2_Tcti_Device_Init (
     tcti_common->locality = 3;
 
 #ifdef __VXWORKS__
-    tcti_dev->fd = open (dev_path, O_RDWR, 0666);
+    tcti_dev->fd = open (dev_path, O_RDWR, (S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP));
```

Comment:
You are missing my point.
If we do not set O_CREAT, then the value that gets passed as `mode` parameter does not do anything, right ?
Thus, I'd put 0 into here.

On a different note: All other posix systems can handle a 3-parameter open as well. So we could get rid of the ifdef here and just use the 3-parameter version for everyone.
Hopefully static code analysis does not dislike this though...

Reply:
For file system, mode parameter is only useful when set O_CREAT. For device, it depends on the implementation. TPM driver of VxWorks will check the mode always, but Linux driver doesn’t. Therefore, it works on Linux without mode parameter, but doesn’t work on VxWorks.
The code is more clean if removed the ifdef, but maybe the value of the mode parameter doesn’t satisfy all of the OS. It will introduce other ifdef.